src/test/java/org/xstefank/check/SkipCheckTest.java

```diff
@@ -0,0 +1,122 @@
+package org.xstefank.check;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.xstefank.api.GitHubAPI;
+import org.xstefank.model.yaml.FormatConfig;
+import org.xstefank.model.yaml.RegexDefinition;
+import org.xstefank.model.yaml.SkipPatterns;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+import static org.xstefank.check.TestUtilities.setUpFormatConfig;
+import static org.xstefank.check.TestUtilities.readJsonNode;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(GitHubAPI.class)
+public class SkipCheckTest {
+
+    private static JsonNode testPullRequestPayload;
+    private static JsonNode testCommitsPayload;
+    private static FormatConfig testFormatConfig = new FormatConfig();
+    private SkipPatterns testSkipPatterns = new SkipPatterns();
+    private RegexDefinition testTitle =  new RegexDefinition();
+    private RegexDefinition testCommit =  new RegexDefinition();
+    private RegexDefinition testDescription =  new RegexDefinition();
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        testPullRequestPayload = readJsonNode(TestUtilities.getTestPullRequestPayloadName());
+        testCommitsPayload = readJsonNode(TestUtilities.getTestCommitsPayloadName());
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullConfigParameter() throws IllegalArgumentException{
+        SkipCheck.shouldSkip(testPullRequestPayload, null);
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullPayloadParameter() throws IllegalArgumentException{
+        SkipCheck.shouldSkip(null, testFormatConfig);
+    }
+
+    @Test
+    public void testSkipByTitleRegexMatch(){
+        testTitle.setPattern(Pattern.compile("test title"));
+        testSkipPatterns.setTitle(testTitle);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        Assert.assertTrue("Method skipByTitle cannot match valid regex", SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
```

Comment:
again "don't forget to update error message when you change the private method somehow"